### PR TITLE
Add assert true hitter rule

### LINKED ISSUE #242: Dumb tests rule
@volodya-lombrozo hi, today I found this test in the project I am working in, I have an idea to have jtcop detect tests that are not acceptable for jtcop.
image

We can discuss a solution here, what do you think?

### PR DESCRIPTION
**Pull request by l3r8yJ**
closes #242 

@volodya-lombrozo take a look, please

---

## PR-Codex overview
This PR focuses on adding line hitter functionality to the code. 

### Detailed summary
- Added `RuleLineHitter` to the list of rules in `RuleCorrectTestCase`
- Added `isLineHitter` method to `AssertionOfJavaParser`, `AssertionOfJUnit`, and `AssertionOfHamcrest`
- Added `assertTrueHitter` and `assertFalseHitter` methods to `AssertionOfJUnitTest`
- Added `containsLineHitter` method to `LineHitterTest`
- Added line hitter related tests

> The following files were skipped due to too many changes: `src/test/java/com/github/lombrozo/testnames/rules/RuleLineHitterTest.java`, `src/main/java/com/github/lombrozo/testnames/rules/RuleLineHitter.java`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Review by volodya-lombrozo — CHANGES_REQUESTED** [review]
@l3r8yJ First of all, great thanks for your contribution! 

I've left several comments to your implementation, but If you don't have enough time to fix them, I would recommend to use puzzles (like you did with the documentation) to show, that all the  "requested changes" will be implemented in the future.

**Comment by l3r8yJ** [comment]
@volodya-lombrozo good suggestions, I'll fix it a bit later

**Comment by l3r8yJ** [comment]
@volodya-lombrozo take a look again, please 

fixed all, maybe didn't notice something

File: src/test/java/com/github/lombrozo/testnames/rules/RuleAssertTrueHitterTest.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                )
+            ).complaints(),
+            Matchers.empty()
+        );
+    }
+
+    @SuppressWarnings("PMD.JUnitTestContainsTooManyAsserts")
+    @Test
+    void checksClassWithHitterCorrectly() {
+        final Assertion assertion = Mockito.mock(Assertion.class);
```
[the remark this answers is not in the document]
Comment (l3r8yJ) [reply src/test/java/com/github/lombrozo/testnames/rules/RuleAssertTrueHitterTest.java:57]: @volodya-lombrozo Yes, I tried using `Assertion.Fake`, but if you make a fake method in the `isLineHitter` method that returns `true`, it breaks the other tests, as the number of complaints increases by one, and it breaks assertions in other tests

I see two options here, mokito or anonymous class, what do you think?

[the remark this answers is not in the document]
Comment (volodya-lombrozo) [reply src/test/java/com/github/lombrozo/testnames/rules/RuleAssertTrueHitterTest.java:57]: @l3r8yJ You can just add a little bit code to the `Assertion.Fake`, something like the next:
```java
 class Fake implements Assertion {
        private final String message;
        private final boolean hitter;

        public Fake() {
            this(UUID.randomUUID().toString());
        }

        public Fake(final String msg) {
           this(msg, false);
        }

        public Fake(final String msg, final boolean hitter) {
            this.message = msg;
            this.hitter = hitter;
        }
    }
```
Then, in the required test you can just invoke `new Assertion.Fake("something", true);

**Comment by l3r8yJ** [comment]
@volodya-lombrozo take a look, please

**Comment by volodya-lombrozo** [comment]
@rultor merge

**Comment by volodya-lombrozo** [comment]
@rultor release, tag is `1.1.0`, title is `1.1.0`

**Pull request merged**